src/cli/java/org/commcare/util/screen/QueryScreen.java

```diff
@@ -143,15 +157,17 @@ public void answerPrompts(Hashtable<String, String> answers) {
 
             // If select question, we should have got an index as the answer which should
             // be converted to the corresponding value
-            if (queryPrompt.isSelectOne() && !answer.isEmpty()) {
-                int choiceIndex = Integer.parseInt(answer);
+            if (queryPrompt.isSelect() && !answer.isEmpty()) {
                 Vector<SelectChoice> selectChoices = queryPrompt.getItemsetBinding().getChoices();
-                if (choiceIndex < selectChoices.size()) {
-                    answer = selectChoices.get(choiceIndex).getValue();
-                } else {
-                    // answer is no longer a valid choice, so clear it out
-                    answer = "";
+                String[] indicesOfSelectedChoices = RemoteQuerySessionManager.extractSelectChoices(answer);
+                ArrayList<String> selectedChoices = new ArrayList<>(indicesOfSelectedChoices.length);
```

Comment:
is it necessary to specify the length here? Just wondering what happens if a choice is invalid, will that result in the final answer having `nulls` in it?

Reply:
it's not the length but initial capacity for the ArrayList. It defaults to 10 if we don't supply any capacity argument here but generally a good practice to supply it explicitly if we have a good estimate to avoid any resizing as the list grows.